packages/typescript-checker/src/typescript-compiler.ts

```diff
@@ -0,0 +1,243 @@
+        getModifiedTime: (fileName: string) => {
+          if (this.fileNameIsBuildInfo(fileName)) {
+            return undefined;
+          }
+          return this.fs.getFile(fileName)?.modifiedTime;
+        },
+        watchFile: (fileName: string, callback: ts.FileWatcherCallback) => {
+          const file = this.fs.getFile(fileName);
+
+          if (file) file.watcher = callback;
+
+          return {
+            close: () => {
+              delete this.fs.getFile(fileName)!.watcher;
+            },
+          };
+        },
+        writeFile: (fileName, data) => {
+          this.fs.writeFile(fileName, data);
+        },
+        watchDirectory: (): ts.FileWatcher => {
+          // this is used to see if new files are added to a directory. Can safely be ignored for mutation testing.
+          return {
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            close() {},
+          };
+        },
+      },
+      (...args) => {
+        const program = ts.createEmitAndSemanticDiagnosticsBuilderProgram(...args);
+        if (this.nodes.size) {
+          return program;
+        }
+        program
+          .getSourceFiles()
+          .filter(filterDependency)
+          .forEach((file) => {
+            this.sourceFiles.set(file.fileName, {
+              fileName: file.fileName,
+              imports: new Set(
+                program
+                  .getAllDependencies(file)
+                  .filter((importFile) => !importFile.includes('/node_modules/') && file.fileName !== importFile)
+                  .flatMap((importFile) => this.resolveFilename(importFile))
+              ),
+            });
+          });
+
+        function filterDependency(file: ts.SourceFile) {
+          if (file.fileName.includes('.d.ts')) return false;
+          if (file.fileName.includes('node_modules')) return false;
+          return true;
+        }
+
+        return program;
+      },
+      (error) => {
+        this.currentErrors.push(error);
+      },
+      (status) => {
+        this.log.debug(status.messageText.toString());
+      },
+      (summary) => {
+        summary.code !== FILE_CHANGE_DETECTED_DIAGNOSTIC_CODE && this.currentTask.resolve();
+      }
+    );
+
+    const compiler = ts.createSolutionBuilderWithWatch(host, [this.tsconfigFile], {});
+    compiler.build();
+    return await this.check([]);
+  }
+
+  public async check(mutants: Mutant[]): Promise<ts.Diagnostic[]> {
+    this.lastMutants.forEach((mutant) => {
+      const file = this.fs.getFile(mutant.fileName);
+      file?.resetMutant();
+    });
+    mutants.forEach((mutant) => {
+      const file = this.fs.getFile(mutant.fileName);
+      file?.mutate(mutant);
+    });
+    await this.currentTask.promise;
+    const errors = this.currentErrors;
+    this.currentTask = new Task();
+    this.currentErrors = [];
+    this.lastMutants = mutants;
+    return errors;
+  }
+
+  public getFileRelationsAsNodes(): Map<string, Node> {
```

Comment:
```suggestion
  public get nodes(): Map<string, Node> {
```

I think this should be property.